clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -104,6 +109,9 @@ public void handleMessage(EntityResponse messageFromServer) {
 
       @Override
       public byte[] createExtendedReconnectData() {
+        for (ReconnectListener reconnectListener : reconnectListeners) {
+          reconnectListener.onHandleReconnect(reconnectData);
```

Comment:
This part where reconnectListener updates reconnect data, (see `ReconnectData`) can also be refactored in a better way, I can use some suggestions there. IMO, should we have a seperate ReconnectMsg refactored and sent to server.